pandas/core/sparse/scipy_sparse.py

```diff
@@ -58,15 +58,7 @@ def _get_label_to_i_dict(labels, sort_labels=False):
             return (d)
 
         def _get_index_subset_to_coord_dict(index, subset, sort_labels=False):
-            def robust_get_level_values(i):
-                # if index has labels (that are not None) use those,
-                # else use the level location
-                try:
-                    return index.get_level_values(index.names[i])
-                except KeyError:
-                    return index.get_level_values(i)
-
-            ilabels = list(zip(*[robust_get_level_values(i) for i in subset]))
+            ilabels = list(zip(*[index._get_level_values(i) for i in subset]))
```

Comment:
I *think* this is OK to do... `_sparse_series_to_coo` is the only caller, and it uses `_get_level_number(x)` to go from maybe labels to numbers.